lib/fluent/plugin/in_github-activities.rb

```diff
@@ -17,98 +17,92 @@
-      @client_threads = []
-      @request_queue = Queue.new
-
-      users_manager_params = {
-        :users    => users,
-        :pos_file => @pos_file,
-      }
-      users_manager = ::Fluent::GithubActivities::UsersManager.new(users_manager_params)
-      users_manager.generate_initial_requests.each do |request|
-        @request_queue.push(request)
+module Fluent
+  module Plugin
+    class GithubActivitiesInput < Fluent::Plugin::Input
+      DEFAULT_BASE_TAG = "github-activity"
+      DEFAULT_CLIENTS = 4
+
+      helpers :thread
+
+      Fluent::Plugin.register_input("github-activities", self)
+
+      config_param :access_token, :string, default: nil, secret: true
+      config_param :users, :array, default: []
+      config_param :users_list, :string, default: nil
+      config_param :include_commits_from_pull_request, :bool, default: false
+      config_param :include_foreign_commits, :bool, default: false
+      config_param :base_tag, :string, default: DEFAULT_BASE_TAG
+      config_param :pos_file, :string, default: nil
+      config_param :clients, :integer, default: DEFAULT_CLIENTS
+      config_param :interval, :integer, default: 1
+
+      def configure(conf)
+        super
+
+        @base_tag = @base_tag.sub(/\.\z/, "")
+        @users += load_users_list
+        @n_clients = [@clients, @users.size].min
+        @interval = @interval * @n_clients
       end
 
-      n_clients.times do
-        @client_threads << Thread.new do
-          crawler_options = {
-            :access_token => @access_token,
-            :watching_users => users,
-            :include_commits_from_pull_request => @include_commits_from_pull_request,
-            :include_foreign_commits => @include_foreign_commits,
-            :pos_file => @pos_file,
-            :request_queue => @request_queue,
-            :default_interval => @interval,
-          }
-          crawler = ::Fluent::GithubActivities::Crawler.new(crawler_options)
-          crawler.on_emit = lambda do |tag, record|
-            router.emit("#{@base_tag}.#{tag}", Engine.now, record)
-          end
+      def start
+        super
+
+        @request_queue = Queue.new
 
-          loop do
-            crawler.process_request
-            sleep(crawler.interval_for_next_request)
+        users_manager_params = {
+          users: @users,
+          pos_file: @pos_file,
+        }
+        users_manager = ::Fluent::Plugin::GithubActivities::UsersManager.new(users_manager_params)
+        users_manager.generate_initial_requests.each do |request|
+          @request_queue.push(request)
+        end
+        @n_clients.times do |n|
+          thread_create("in_github_activity_#{n}".to_sym) do
+            crawler_options = {
+              access_token: @access_token,
+              watching_users: @users,
+              include_commits_from_pull_request: @include_commits_from_pull_request,
+              include_foreign_commits: @include_foreign_commits,
+              pos_file: @pos_file,
```

Comment:
Can we pass a storage plugin instance instead of pos_file here like as https://github.com/reevoo/fluent-plugin-systemd/pull/31/commits/7f379cbc17fcc804a6eda75aa6ac746b9fe3a050?